warehouse/query-core/src/main/java/datawave/query/iterator/logic/DeferredIterator.java

```diff
@@ -0,0 +1,272 @@
+package datawave.query.iterator.logic;
+
+import com.google.common.collect.TreeMultimap;
+import datawave.query.iterator.NestedIterator;
+import datawave.query.iterator.Util;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class DeferredIterator {
```

Comment:
Also, this is not actually an iterator.  This is a utility class.  Perhaps DeferredIteratorUtil?